### ISSUE TITLE
RADIUS SERVER IP ADDITION IN DB

### ISSUE THREAD

**Issue by ztom1991** [zentyal/zentyal#2105]
**Describe the bug**
IN THE RADIUS MODULE, WHEN ADMINISTRATOR ADDS RADIUS CLIENTS FOR AUTHENTICATION, SERVER TAKES DESCRIPTION, IP ADDRESS, SUBNET MASK AND PRESHARED SECRET AS INPUT. WHEN CONFIGURATION IS SAVED, SERVERS ADDS IP ADDRESS AGAINST DESCRIPTION AND DOES NOT MAKE 'IP ADDR' FIELD IN RADIUS.CONF FILE. THEREFORE, IP ADDRESS CANNOT BE AUTHENTICATED.

**To Reproduce**
1. GO TO TERMINAL AND EXECUTE COMMAND tail -f /var/log/freeradius/radius.log
2. Go to 'RADIUS TAB'
3. Click on 'ADD NEW'
4. ADD REFQUIRED DATA
5. CLICK ADD
6. GO BACK TO TERMINAL AND SEE ERROR "NO IPADDR OR IPV4ADDR OR IPV6ADDR FIELD FOUND IN CLIENT 192.168.X.X/32. PLEASE FIX YOUR CONFIGURATION"
7. THE ERROR CAN ALSO BE VALIDATED BY CHECKING RADIUSD.CONF FILE IN FREERADIUS INSTALLATION DIRECTORY

**Expected behavior**
Radius client gets authenticated solely on the basis of user credentials and certificate. No ip address is checked.

**Screenshots**
will be shared on demand.

**Zentyal OS (please complete the following information):**
 - Version: [6.2]
 - Version of the module which has the bug [ freeradius 3.0]
 - Modules installed [ALL]

**Comment by djoven89** [comment]
Hi @ztom1991 ,

Thank you for reporting this. We have reproduced the issue and it was added to the roadmap.

In the tests we did in our lab, the following code seems to fix the issue. Could you confirm it by modifying the stub located at `/usr/share/zentyal/stubs/radius/clients.conf.mas`, please?

```
% foreach my $client (@clients) {
client <% $client->{'name'} %> {
    ipaddr = <% $client->{'ipaddr'} %>
    secret = <% $client->{'secret'} %>
    require_message_authenticator = no
    nastype = other
}
% }
```

**NOTE:** Do not forget to restart the Radius module (`sudo zs radius restart`).

**Comment by ztom1991** [comment]
Hi djoven89, Thank you for reproducing this issue and helping out. I shall modify code and confirm it by monday. RegardsTaimur ZahidOn 23-Sep-2022, at 4:28 PM, djoven89 ***@***.***> wrote:﻿
Hi @ztom1991 ,
Thank you for reporting this. We have reproduced the issue and it was added to the roadmap.
In the tests we did in our lab, the following code seems to fix the issue. Could you confirm it by modifying the stub located at /usr/share/zentyal/stubs/radius/clients.conf.mas, please?
% foreach my $client ***@***.***) {
client <% $client->{'name'} %> {
    ipaddr = <% $client->{'ipaddr'} %>
    secret = <% $client->{'secret'} %>
    require_message_authenticator = no
    nastype = other
}
% }

NOTE: Do not forget to restart the Radius module (sudo zs radius restart).

**Comment by ztom1991** [comment]
Hi djoven89,

Thank you for the quick fix. I have applied recommended changes and the issue has been resolved. The freeradius clients.conf file now displays client name set from ZENTYAL web gui. 

However I wanted to ask another question. ZENTYAL accepts credentials from any client no matter what the IP address is. This is the SSO (single sign on) feature I believe. Is it possible to bind user credentials against IP address so that a specific IP address can only use a specific set of credentials? 

Thanks

Regards
